torch/utils/_config_module.py

```diff
@@ -26,19 +27,53 @@ class Config:
 
     This configs must be installed with install_config_module to be used
 
+    Environment Variables:
+        These are interpreted to be true / false variables.
+
     Arguments:
         justknob: the name of the feature / JK. In OSS this is unused.
         default: is the value to default this knob to in OSS.
+        env_name_force: The environment variable to read that is a FORCE
+            enviornment variable. I.e. it overrides everything
+        env_name_default: The environment variable to read that changes the
+            default behaviour. I.e. user overrides take preference.
+
+
+    Evaluation Priority (hightest to lowest):
+        env_name_force
+        user_override
+        env_name
+        justknob
+        default
+
+    i.e. if set, env_name_force would override a user override, but a user override will override a JK.
     """
 
     default: bool = True
     justknob: Optional[str] = None
+    env_name_literal: bool = False
+    env_name_default: Optional[str] = None
+    env_name_force: Optional[str] = None
 
 
 # Types saved/loaded in configs
 CONFIG_TYPES = (int, float, bool, type(None), str, list, set, tuple, dict)
 
 
+def _read_env_variable(name: str) -> Optional[bool]:
+    if (env := os.getenv(name)) is not None:
+        env = env.upper()
+        if env in ("1", "TRUE"):
+            return True
+        if env in ("0", "FALSE"):
```

Comment:
What about blank envvar content? Also, note that this parsing is incompatible with most of the idiomatic os.getenv parsing that is currently in use, e.g.,

```
def _get_tristate_env(name: str) -> Optional[bool]:
    value = os.environ.get(name)
    if value == "1":
        return True
    if value == "0":
        return False
    return None
```

Reply:
There is a small (but not zero) amount of usage of FALSE + TRUE. I'll just use _get_tristate_env though :).
```
 % grep \'TRUE torch -r
torch/utils/cpp_extension.py:    if os.environ.get('TORCH_DONT_CHECK_COMPILER_ABI') in ['ON', '1', 'YES', 'TRUE', 'Y']:
torch/utils/cpp_extension.py:            os.environ.get('TORCH_DONT_CHECK_COMPILER_ABI') not in ['ON', '1', 'YES', 'TRUE', 'Y'] and
clr@clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fed ~/pytorch
 % grep \'TRUE torch -r
torch/utils/cpp_extension.py:    if os.environ.get('TORCH_DONT_CHECK_COMPILER_ABI') in ['ON', '1', 'YES', 'TRUE', 'Y']:
torch/utils/cpp_extension.py:            os.environ.get('TORCH_DONT_CHECK_COMPILER_ABI') not in ['ON', '1', 'YES', 'TRUE', 'Y'] and
clr@clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fed ~/pytorch
 % grep \"TRUE torch -r
torch/_inductor/cpp_builder.py:            os.environ["KMP_DUPLICATE_LIB_OK"] = "TRUE"
torch/_inductor/cpp_builder.py:                os.environ["KMP_DUPLICATE_LIB_OK"] = "TRUE"
torch/utils/_config_module.py:        if env in ("1", "TRUE"):
clr@clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fed ~/pytorch
 % grep \'FALSE torch -r
clr@clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fedora-MJ0J4X74clr-fed ~/pytorch
 % grep \"FALSE torch -r
torch/csrc/jit/frontend/ir_emitter.cpp:      std::strcmp(enable_env, "FALSE") == 0 ||
torch/fx/passes/_tensorify_python_scalars.py:        if env in ("0", "FALSE"):
torch/utils/_config_module.py:        if env in ("0", "FALSE"):
```

Reply:
Ended up reimplementing it, since _get_tristate_env is defined in config.

Reply:
But I think this now follows what get_tristate_env does, and just returns none for any env variables it doesn't understand.